service/src/main/java/com/ge/predix/acs/zone/resolver/SpringSecurityZoneResolver.java

```diff
@@ -38,16 +41,29 @@ public static ZoneEntity getZoneEntity() {
                     .getAuthentication();
             throw new InvalidACSRequestException("The zone '" + zoneAuth.getZoneId() + "' does not exist.");
         }
-
         return result;
     }
     
     @Override
     public ZoneEntity getZoneEntityOrFail() {
-        return getZoneEntity();
+        ZoneEntity result = null;
+        try {
+        result = getZoneEntity();
+        } catch (InvalidACSRequestException e) {
+            LOGGER.debug(e.getMessage());
+            return null;
```

Comment:
This change will now return null to ```PrivilegeManagementServiceImpl``` which will throw null point exception.
I think it might be better to create a custom exception for when zone does not exist and create an exception handler in a class annotated with ```@ControllerAdvice```